manta-js/package/src/utils.ts

```diff
@@ -1,49 +1,283 @@
 
-export const NATIVE_TOKEN_ASSET_ID = '1';
-
-/// MantaUtilities class
-export class MantaUtilities {
-
-  /// Returns the public balance associated with an account for a given AssetId.
-  static async getPublicBalance(api:ApiPromise, assetId: BN, address:Address): Promise<BN | null> {
-    try {
-      if (assetId.toString() === NATIVE_TOKEN_ASSET_ID) {
-        const nativeBalance: any = await api.query.system.account(address);
-        return new BN(nativeBalance.data.free.toString());
-      } else {
-        const assetBalance: any = await api.query.assets.account(assetId, address);
-        if (assetBalance.value.isEmpty) {
-          return new BN(0);
-        } else {
-          return new BN(assetBalance.value.balance.toString());
-        }
-      }
-    } catch (e) {
-      console.log('Failed to fetch public balance.');
-      console.error(e);
-      return null;
-    }
-  }
+/// Convert a private address to JSON.
+export function convertZkAddressToJson(address: string) {
+  const bytes = base58Decode(address);
+  return JSON.stringify({
+    receiving_key: Array.from(bytes),
+  });
+}
+
+export async function mapPostToTransaction(
+  palletName: PalletName,
+  api: ApiPromise,
+  post: any,
+): Promise<SubmittableExtrinsic<'promise', any>> {
+  if (palletName === 'mantaPay') {
+    const sources = post.sources.length;
+    const senders = post.sender_posts.length;
+    const receivers = post.receiver_posts.length;
+    const sinks = post.sinks.length;
 
-  /// Executes a public transfer.
-  static async publicTransfer(api:ApiPromise, assetId: BN, amount: BN, destinationAddress: Address, senderAddress:Address, polkadotSigner:Signer): Promise<void> {
-    api.setSigner(polkadotSigner);
-    try {
-      const assetIdArray = bnToU8a(assetId, {bitLength: 256});
-      const amountBN = amount.toArray('le', 16);
-      const tx = await api.tx.mantaPay.publicTransfer(
-        { id: assetIdArray, value: amountBN },
-        destinationAddress
+    if (sources == 1 && senders == 0 && receivers == 1 && sinks == 0) {
+      const mintTx = await api.tx.mantaPay.toPrivate(post);
+      return mintTx;
+    } else if (sources == 0 && senders == 2 && receivers == 2 && sinks == 0) {
+      const privateTransferTx = await api.tx.mantaPay.privateTransfer(post);
+      return privateTransferTx;
+    } else if (sources == 0 && senders == 2 && receivers == 1 && sinks == 1) {
+      const reclaimTx = await api.tx.mantaPay.toPublic(post);
+      return reclaimTx;
+    } else {
+      throw new Error(
+        'Invalid transaction shape; there is no extrinsic for a transaction' +
+          `with ${sources} sources, ${senders} senders, ` +
+          ` ${receivers} receivers and ${sinks} sinks`,
       );
-      await tx.signAndSend(senderAddress);
-    } catch (e) {
-      console.log('Failed to execute public transfer.');
-      console.error(e);
     }
+  } else if (palletName === 'mantaSBT') {
+    // This method is temporarily unavailable
```

Comment:
is this comment still true?

Reply:
Discussed with qihuang, SBT-related interfaces only return posts, there is no need to format posts into transactions in the SDK, dApp can do this format

Reply:
I commented out this code for now